hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -81,7 +89,11 @@ class KubernetesClient {
         this.useNodeNameAsExternalAddress = useNodeNameAsExternalAddress;
         this.servicePerPodLabelName = servicePerPodLabelName;
         this.servicePerPodLabelValue = servicePerPodLabelValue;
-        this.apiProvider = buildKubernetesApiUrlProvider();
+        this.clusterTopologyIntentTracker = clusterTopologyIntentTracker;
+        this.apiProvider =  buildKubernetesApiUrlProvider();
+        this.stsName = extractStsName();
+        this.stsMonitorThread = clusterTopologyIntentTracker != null
+                ? new Thread(new StsMonitor(), "hz-sts-monitor") : null;
```

Comment:
```suggestion
                ? new Thread(new StsMonitor(), "hz-k8s-sts-monitor") : null;
```
Adding k8s to the name can help to understand what it does easily in thread dumps.